test/fw/ptl/lib/ptl_server.py

```diff
@@ -0,0 +1,6087 @@
+                         mode)
+        return mode
+
+    def add_expect_action(self, name=None, action=None):
+        """
+        Add an action handler to expect. Expect Actions are
+        custom handlers that are triggered when an unexpected
+        value is encountered
+
+        :param name: Action name
+        :type name: str or None
+        :param action: Action to add
+        """
+        if name is None and action.name is None:
+            return
+        if name is None and action.name is not None:
+            name = action.name
+
+        if not self.actions.has_action(name):
+            self.actions.add_action(action, self.shortname)
+
+    def set_attributes(self, a={}):
+        """
+        set server attributes
+
+        :param a: Attribute dictionary
+        :type a: Dictionary
+        """
+        super(Server, self).set_attributes(a)
+        self.__dict__.update(a)
+
+    def isUp(self, inst, max_attempts=None):
+        """
+        returns ``True`` if server is up and ``False`` otherwise
+        """
+        if max_attempts is None:
+            max_attempts = self.ptl_conf['max_attempts']
+        if self.has_snap:
+            return True
+        i = 0
+        op_mode = self.get_op_mode()
+        if ((op_mode == PTL_API) and (self._conn is not None)):
+            self._disconnect(self._conn, force=True)
+        while i < max_attempts:
+            rv = False
+            try:
+                if op_mode == PTL_CLI:
+                    self.status(SERVER, level=logging.DEBUG, logerr=False)
+                else:
+                    c = self._connect(self.hostname)
+                    self._disconnect(c, force=True)
+                return True
+            except (PbsConnectError, PbsStatusError):
+                # if the status/connect operation fails then there might be
+                # chances that server process is running but not responsive
+                # so we wait until the server is reported operational.
+                rv = self._isUp(inst)
+                # We really mean to check != False rather than just "rv"
+                if str(rv) != 'False':
+                    self.logger.warning('Server process started' +
+                                        'but not up yet')
+                    time.sleep(1)
+                    i += 1
+                else:
+                    # status/connect failed + no server process means
+                    # server is actually down
+                    return False
+        return False
+
+    def start(self, args=None, launcher=None):
+        """
+        Start the PBS server
+
+        :param args: Argument required to start the server
+        :type args: str
+        :param launcher: Optional utility to invoke the launch of the service
+        :type launcher: str or list
+        """
+        if args is not None or launcher is not None:
+            rv = super(Server, self)._start(inst=self, args=args,
```

Comment:
Here no need of super. You can just call self._start().
Or update the method name in PBSService from _start to start.